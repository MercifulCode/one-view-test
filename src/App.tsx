import * as React from "react";
import { Provider } from "react-redux";
import "./App.css";
import { PostTable } from "./components/PostTable";
import { UserTable } from "./components/UserTable";
import { store } from "./Store";

export interface IAppState {
  currentUser: IUserResponse | null;
}

export class App extends React.Component<{}, IAppState> {
  constructor(props: any) {
    super(props);
    this.state = {
      currentUser: null,
    };
  }

  public render() {
    const { currentUser } = this.state;

    return (
      <Provider store={store}>
        <div
          className="App"
          style={{
            display: "flex",
            flexDirection: "column",
            justifyContent: "center",
          }}
        >
          <UserTable onClick={this.onClickCallback}></UserTable>
          {currentUser ? this.renderPostTable(currentUser) : null}
        </div>
      </Provider>
    );
  }

  protected renderPostTable(currentUser: IUserResponse) {
    return (
      <div>
        <br />
        {currentUser.name}'s Posts:
        <PostTable currentUser={currentUser} />
      </div>
    );
  }

  protected onClickCallback = (user: IUserResponse) => {
    this.setState({
      currentUser: user,
    });
  };
}

export default App;
