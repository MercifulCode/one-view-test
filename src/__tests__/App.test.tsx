import { App } from "../App";
import { shallow } from "enzyme";

describe("App", () => {
  it("Should render the app.", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
