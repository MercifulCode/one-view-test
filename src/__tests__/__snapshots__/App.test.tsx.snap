// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render the app. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="App"
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Connect(UserTableClass)
      onClick={[Function]}
    />
  </div>
</Provider>
`;
